### ISSUE TITLE
SIA-R111 and SIA-R113 improvements

### ISSUE THREAD

**Issue by Jym77** [Siteimprove/alfa#1593]
- [ ] Make inline targets inapplicable. 
- [ ] Make invisible targets inapplicable.
- [ ] Make `<area>` elements inapplicable (due to the often weird shape).
- [ ] Improve UA controlled exception to keep those whose size has been modified by author.
- [ ] Consider `<label>` as part of the size of their labelled element; this is probably OK until rpevious point is done, since in most of the case the labellable element would be an `<input>` that is currently ignored. Should we merge rectangles? Probably only if they are "close enough", but how to detect that? Implicit labels should be good, for explicit ones we need to have some heuristic of "closeness".
- [ ] Optimize collision detection
- [ ] Factor duplicated `Outcome` namespaces of R111 and R113

**Comment by rcj-siteimprove** [comment]
Even though inline and invisible targets should be inapplicable, they should still be regarded when calculating spacing of an applicable target. In other words an applicable target must still not be too close to e.g. a link in a text paragraph or some invisible button. Is that correctly understood @Jym77? What about the `<area>` elements, should they also be regarded when calculating spacing of an applicable target?

**Comment by Jym77** [comment]
Yes, all interactive content should be considered for spacing (main use case: touchscreen with low granularity where the contact point might cover more than one element).

**Comment by MaksimShakavin** [comment]
Hello @rcj-siteimprove @Jym77 , 
I created a bug https://github.com/Siteimprove/alfa/issues/1639 which is probably connected to this issue. If you think that it duplicates the topics covered in the issue of may be already fixed but not released, feel free to close it.
